test/test.js

```diff
@@ -5,111 +5,598 @@ suite('GaiaHeader', function() {
+      this.dom.innerHTML = '<gaia-header action="back"><h1>foo</h1></gaia-header>';
+
+      var el = this.dom.firstElementChild;
+      var h1 = el.querySelector('h1');
+
+      afterNext(el, 'runFontFit').then(() => {
+        this.fontFit.reset();
+        h1.textContent = 'bar';
+
+        return afterNext(el, 'runFontFit').then(() => {
+          sinon.assert.calledOnce(this.fontFit);
+        });
+      }).then(done, done);
+    });
+
+    test('It runs font-fit when `action` changes', function(done) {
+      this.dom.innerHTML = '<gaia-header action="back"><h1>foo</h1></gaia-header>';
+      var el = this.dom.firstElementChild;
+
+      afterNext(el, 'runFontFit').then(() => {
+        this.fontFit.reset();
+
+        // Removing the action should
+        // hide the action-button, so
+        // font-fit needs to be run.
+        el.action = null;
+
+        return afterNext(el, 'runFontFit').then(() => {
+          sinon.assert.calledOnce(this.fontFit);
+        });
+      }).then(done, done);
+    });
+
+    test('It runs font-fit when a button is hidden', function(done) {
+      this.dom.innerHTML = `<gaia-header action="back">
+        <h1>foo</h1>
+        <button></button>
+      </gaia-header>`;
+
+      var el = this.dom.firstElementChild;
+      var button = el.querySelector('button');
+
+      afterNext(el, 'runFontFit').then(() => {
+        this.fontFit.reset();
+
+        // Hiding a button triggers
+        // the mutation observer
+        button.hidden = true;
+
+        return afterNext(el, 'runFontFit').then(() => {
+          sinon.assert.calledOnce(this.fontFit);
+        });
+      }).then(done, done);
+    });
+
+    test('It runs when a button\'s classList changes', function(done) {
+      this.dom.innerHTML = `<gaia-header action="back">
+        <h1>foo</h1>
+        <button></button>
+      </gaia-header>`;
+
+      var el = this.dom.firstElementChild;
+      var button = el.querySelector('button');
+
+      afterNext(el, 'runFontFit').then(() => {
+        this.runFontFit.reset();
+
+        // Hiding a button triggers
+        // the mutation observer
+        button.classList.add('hidden');
+
+        return afterNext(el, 'runFontFit').then(() => {
+          sinon.assert.calledOnce(this.runFontFit);
+        });
+      }).then(done, done);
+    });
+  });
+
+  suite('title-start and title-end attributes', function() {
+    var h1, el;
+
+    function insertHeader(container, attrs, done) {
+      done = done || function() {};
+      attrs = attrs || {};
+
+      var start = 'titleStart' in attrs ? 'title-start="' +  attrs.titleStart + '"' : '';
+      var end = 'titleEnd' in attrs ? 'title-end="' +  attrs.titleEnd + '"': '';
+
+      container.innerHTML = `
+        <gaia-header action="back" ${start} ${end}>
+          <h1>Title</h1>
+        </gaia-header>
+      `;
+
+      el = container.firstElementChild;
+      h1 = el.querySelector('h1');
+
+      return afterNext(el, 'runFontFit').then(done);
```

Comment:
I'd just return the result of afterNext and let the caller add `.then(done)` if necessary